mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp

```diff
@@ -55,6 +58,52 @@ void createAttnTuningRangeBF(TuningParamSet *newSpace, AttentionOp attnOp,
   }
 }
 
+auto getOptinalSplitKFactors(RockGemmWrapperInterface gemmOp,
+                             int32_t gemmMPerBlock, int32_t gemmNPerBlock,
+                             int32_t gemmKPerBlock, int32_t kPack) {
+  auto info = PopulateParamsInfo::fromOp(gemmOp);
+  if (not info.gemmAType.isF32()) {
+    return std::vector<int64_t>{1};
+  }
+
+  struct LocalData {
+    int64_t splitKFactor{};
+    double workImbalance{};
+  };
+
+  const auto numCUs = gemmOp.getNumCU().value();
+  std::vector<LocalData> factors{};
+  for (int32_t splitKFactor = 1; splitKFactor < 100; ++splitKFactor) {
+    auto kPerBlock = math_util::lcm(gemmKPerBlock * kPack, splitKFactor);
```

Comment:
LLVM style limits auto to ones that're obvious from context (ex, you'd be repeating the type name from a `cast<>` or where the underlying type is wordy and not that important. I'd like the integer types written out explicitly here.

Reply:
Understood!